azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1098,6 +1113,86 @@ private void ajaxCheckForWritePermission(final Project project, final User user,
     ret.put("hasWritePermission", hasPermission(project, user, Type.WRITE));
   }
 
+  /**
+   * Set if a flow is locked.
+   *
+   * @param project the project for the flow.
+   * @param ret the return value.
+   * @param req the http request.
+   * @throws ServletException
+   */
+  private void ajaxSetFlowLock(final Project project,
```

Comment:
Yes, the flow trigger would be paused. The flow trigger would be resumed after the flow is unlocked. This helps give a visual clue that the flow is not currently being executed. Alternatively we could have added a "locked" column. The plan is to add a tool tip over the disabled resume button, which can mention that the flow is locked.

 Flows using the original schedule would not run, but locking will not remove the schedule, in case they want to use the same schedule and SLA options after the flow is unlocked again -- they would not need to set the schedule.

Would you prefer to unschedule locked flows? Or not pause flow triggers?

Reply:
Ideally, schedule and flow trigger should work in the same way. If we have the plan to consolidate them together in the future, then it might be easier to do so. Adding a "locked" column seems also a good idea. For now, you can keep your existing implementation and put those improvements in the todo list.